src/service/viewer-impl.js

```diff
@@ -221,6 +221,10 @@ export class Viewer {
     this.performanceTracking_ = this.params_['csi'] === '1';
     dev.fine(TAG_, '- performanceTracking:', this.performanceTracking_);
 
+    this.shareTrackingIncomingFragment_ = this.params_['share-tracking'];
```

Comment:
I feel this param doesn't deserve a seat in this file.

You can move the logic to the caller (the extension).